internal/agent/list.go

```diff
@@ -12,41 +12,86 @@ import (
 var agentListStyle = lipgloss.NewStyle().Margin(1, 2)
 
 type AgentListModel struct {
-	agentList   list.Model
-	agentLoader tea.Cmd
+	agentList        list.Model
+	agentCurrentPage int
+	agentPerPage     int
+	agentLastPage    int
+	agentsLoading    bool
+	agentLoader      func(int, int) tea.Cmd
 }
 
-func NewAgentList(loader tea.Cmd) AgentListModel {
+func NewAgentList(loader func(int, int) tea.Cmd, page, perpage int) AgentListModel {
 	l := list.New(nil, NewDelegate(), 0, 0)
 	l.Title = "Buildkite Agents"
 
 	return AgentListModel{
-		agentList:   l,
-		agentLoader: loader,
+		agentList:        l,
+		agentCurrentPage: page,
+		agentPerPage:     perpage,
+		agentLoader:      loader,
 	}
 }
 
+func (m *AgentListModel) appendAgents() tea.Cmd {
+	// Set agentsLoading
+	m.agentsLoading = true
+	// Set a status message and start the agentList's spinner
+	startSpiner := m.agentList.StartSpinner()
+	setStatus := m.agentList.NewStatusMessage(("Fetching more agents"))
+	// Fetch and append more agents
+	appendAgents := m.agentLoader(m.agentCurrentPage, m.agentPerPage)
+	return tea.Sequence(appendAgents, tea.Batch(startSpiner, setStatus))
```

Comment:
I think the arguments should be backwards here right? Sequence runs each Cmd in order until they are finished, so it won't show the spinner/status until its finished loading agents?

Reply:
I've flipped them around, oddly was getting more consistent popup/loading with this ordering, though I think its totally YMMV

Reply:
hmm could be my understanding then. perhaps thats involved with the intermittent issues ive seen in #188 🤔